Support custom CSV delimiters

blp opened the pull request:
TPC-H uses `|` instead of `,` and I'm sure that other datasets use other characters too.

ryzhyk commented on crates/adapters/src/adhoc/table.rs:
```diff
@@ -49,6 +50,7 @@ pub const fn datafusion_arrow_serde_config() -> &'static SqlSerdeConfig {
         date_format: DateFormat::String("%Y-%m-%d"),
         decimal_format: DecimalFormat::String,
         variant_format: VariantFormat::JsonString,
+        delimiter: CsvDelimiter::DEFAULT,
```
It's odd that this needs to be part of `SqlSerdeConfig`, which is only supposed to contain properties related to deserializing individual data types, not how the parses handles the envelope. Is it possible to refactor this to not have `delimiter` in `SqlSerdeConfig`?

blp replied:
OK, I'll figure it out. I didn't understand what was happening with SqlSerdeConfig.

blp replied:
Fixed.

mihaibudiu commented on web-console/src/lib/services/manager/schemas.gen.ts:
```diff
@@ -495,21 +495,35 @@ snapshot.  The condition must be a valid SQL Boolean expression that can be used
-in Feldera SQL. It impacts the performance of the connector, since data must be sorted
-before pushing it to the pipeline; therefore it is not recommended to use this
-settings for tables without \`LATENESS\`.`,
+table rows are ingested in the timestamp order, respecting the
```
I thought I saw this in a prior commit

ryzhyk reviewed (commented):
Other comments:
* you probably want to update `docs/formats/csv.md` as well.
* another useful option is to enable/disable CSV headers, but doesn't have to be in this PR.

blp commented:
> Other comments:
> 
>     * you probably want to update `docs/formats/csv.md` as well.

Updated.

>     * another useful option is to enable/disable CSV headers, but doesn't have to be in this PR.

I added this but it's a little weird, because the csv writer can't write headers in all the cases where it can write records. See "However, if `has_headers` were enabled..." just below the code example at https://docs.rs/csv/latest/csv/struct.Writer.html#structs. So, I didn't enable headers for csv output, only for input.

mihaibudiu commented:
Can I put this in the queue?
